### ISSUE TITLE
Could not move to the next location (no location selected) console error

### ISSUE THREAD

**Issue by aylmaao** [Regalis11/Barotrauma#7310]
- [ ] I have searched the issue tracker to check if the issue has already been reported.
- [ ] I found this issue but it's closed and I'm not sure how relevant it is https://github.com/Regalis11/Barotrauma/issues/5553

I had already moved from the location it shows I am to the location directly below it (I activated the beacon along the way) but I got a console error and was put back to the location I started from.

**Steps To Reproduce**
I attached the save file

**Version**
Among the ancients v0.15.12.0
- Edit: actually I think it was version v0.15.13.0

Before this happened I suspect there was also a bug where I had completed a scan mission, the one where you place the beacon in 3 locations, but once going to the next location it said mission failed. I had the beacon stored in a storage compartment on the ship (not in a crate). But I'm new to the game so I may have made a mistake somewhere.
That happened in the area just before this one
[ayylmao.zip](https://github.com/Regalis11/Barotrauma/files/7534304/ayylmao.zip)
![ayylmao bug 1](https://user-images.githubusercontent.com/94312689/141692624-5094486a-b49c-41d7-8c55-3febb2b65310.png)
![ayylmao bug 2](https://user-images.githubusercontent.com/94312689/141692629-b2c8c057-c173-4b5a-a203-93cbba269746.png)
.

**Comment by pakvg85** [comment]
It seems this fatal error occurs when following circumstances are met:
1. No mission is chosen for a route.
2. There is a beacon on route.
Does not matter if you activate the beacon or not - the error will occur.

![barotrauma - successful route](https://user-images.githubusercontent.com/45126308/175511492-f3604a1b-1eda-4803-b358-76348b034a91.png)

Devs, could you please repeat & fix this error?

**Comment by Regalis11** [comment]
Thank you for the extra info!

We would of course like to fix this error as soon as possible, but unfortunately we still haven't been able to find a way to reproduce it at our end. I just gave a test at entering a level with a beacon station with no mission selected, in two different levels, but didn't get any kind of errors:
![image](https://user-images.githubusercontent.com/14351070/175915541-45f29553-f5e2-41d8-b1e2-3f5d87f9e75b.png)
![image](https://user-images.githubusercontent.com/14351070/175915634-63f2c125-0a74-43f3-9786-257989dff4da.png)

Would it be possible to provide us the save file the issue appeared in, in case we're able to reproduce the issue there?

**Comment by pakvg85** [comment]
Should I attach the save file only after error occur or any save file from my campaign will fit?
My current save file is in atttachement (not after the error).
[SaveN.zip](https://github.com/Regalis11/Barotrauma/files/8991859/SaveN.zip)
.

**Comment by Regalis11** [comment]
Thank you! I would imagine that I could reproduce the issue with any save from that campaign, if it is something that occurs every time in that specific level. I could not load that save though, it throws the error "unexpected game mode" when trying to load it. Maybe the file's gotten corrupted some how? Is that a normal .zip archive or another type of file whose file extension you changed to allow attaching it here?

**Comment by pakvg85** [comment]
It is a normal .ZIP archive. Could it be because I am using a custom submarine?
Attached it in case it could affect loading. Also, Jovian radiation is turned off.
[LocalMods.zip](https://github.com/Regalis11/Barotrauma/files/8998513/LocalMods.zip)
.

**Comment by pakvg85** [comment]
I repeated the situation. Prerequisites are the same:
1. Starting point is empty point (absent of outpost)
2. Final point is an outpost.
3. There is a beacon that should be activated on a route
![map before mission failed](https://user-images.githubusercontent.com/45126308/176956035-094aec69-91be-48ea-81b6-b8e5deb162c8.png)
.
In attachement there are console screenshots, map screenshots and 2 save files - when mission is started and after mission was failed.

**Comment by pakvg85** [comment]
Also I just understood what user "aylmaao" had meant when he said 
> there was also a bug where I had completed a scan mission, the one where you place the beacon in 3 locations, but once going to the next location it said mission failed. I had the beacon stored in a storage compartment on the ship (not in a crate). But I'm new to the game so I may have made a mistake somewhere. That happened in the area just before this one

Situation: I accepted "Scan ruins" mission, successfully placed all 3 scanners on yellow spots inside alien ruins. After third scanner finished scanning, I received message "mission complete". Then I took all 3 scanners back to my submarine. But when I arrived to destination point, the "scan ruins" mission was failed. I do not understand why.

**Comment by pakvg85** [comment]
I tried to launch the save file from another PC and the error did not appear when I reached the end of level. I will let know if it appears again.

**Comment by pakvg85** [comment]
> issue where my sub started with everything heavily damaged at the start of the location
My theory is:
When you press "Start mission", a new level starts constructing - environment, enemies, wrecks, ruins etc. While level is being constructed, your submarine is put into this enviroment in the very beginning and all systems start living as usual - reactor consumes fuel, batteries are charging (or discharging, if reactor is off and load is >0).
Those systems deteriorate all the time while level is being constructed, but there are no bots to repair them because bots are not yet loaded.

I think it could be easily be fixed - just put player's submarine into enviroment in the very end of level constructing.

**Comment by pakvg85** [comment]
> I've also had this problem on my first campaign, with the same reproduction methods as the others

Could you check integrity of game files? I did this and there were 2 damaged files. After fixing those 2 files the error with beacons was gone.
![barotrauma - check integrity 1](https://user-images.githubusercontent.com/45126308/178073866-ddadc3f1-1418-4385-84d1-8af3db9a1bb7.png)
?

**Comment by TaryntEssrog** [comment]
Hi, 

I was directed by a user on discord to post here that I am experiencing the same bug. They said that checking the integrity fixed two corrupted files and fixed the problem, but I didn't have any corrupted files and the problem persists. Same situation with traveling from a natural formation to an outpost and past a beacon.

Attached are images of the debug console and the saved game file.

![image](https://user-images.githubusercontent.com/110116144/181285311-22d98a84-6f28-4922-8c64-a6cf0ecb2298.png)
![image](https://user-images.githubusercontent.com/110116144/181285411-3e517ea3-9d24-4980-9f74-d2f049b5db2f.png)
![image](https://user-images.githubusercontent.com/110116144/181285452-bd8457c2-abf3-4cef-8430-0fb26b8c0bf5.png)
[save file.zip](https://github.com/Regalis11/Barotrauma/files/9200457/save.file.zip)

**Comment by eTATr** [comment]
我也遇到了这个问题，游戏版本0.19.14.0，按照@pakvg85的方法验证完整性之后发现了5个文件错误。第一次验证完文件后我又重新按照“Lua For Barotrauma"的说明启动了这个MOD，但还是无法到达下一个站点。第二次验证完文件后没有启动”Lua For Barotrauma“，成功到达了下一个站点。

**Comment by Instant2116** [comment]
[Unstable 0.20.09.0] Same error. I chose rote to the place without outpost, pass the route, find out that end point is outpost docking port (where it get from?), docked, then get error massage and teleported to different empty point from other unselected route. 
[error msg.txt](https://github.com/Regalis11/Barotrauma/files/10219767/error.msg.txt)

**Comment by Regalis11** [comment]
Duplicate of https://github.com/Regalis11/Barotrauma/issues/10772. The issue should be fixed, but the fix is unfortunately not available in the public build yet.
